independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/ClientSerialisers.java

```diff
@@ -104,10 +103,10 @@ public static Buffer invokeClientWriter(Entity<?> entity, Object entityObject, C
 
         if (writer.isWriteable(entityClass, entityType, entity.getAnnotations(), entity.getMediaType())) {
             if (writerInterceptors == null) {
-                ByteArrayOutputStream baos = new ByteArrayOutputStream();
+                VertxBufferOutputStream out = new VertxBufferOutputStream();
                 writer.writeTo(entityObject, entityClass, entityType, entity.getAnnotations(),
-                        entity.getMediaType(), headerMap, baos);
-                return Buffer.buffer(baos.toByteArray());
+                        entity.getMediaType(), headerMap, out);
+                return out.getBuffer();
```

Comment:
I'm probably missing a lot, but isn't it still keeping it all in a single buffer before sending?

Reply:
Before these changes, we were storing the large file in a byte array (ByteArrayOutputStream). 
With these changes, we will directly use the buffer which works similarly than BufferedOutputStream.

Reply:
What Michal is asking IIUC, is whether all the data is still in memory or not.

Reply:
Let me try and rephrase: If one is to send a 20GB file back, how much data is in memory during the sending operation? It is a constant number, or does it increase with file size?

Asking because, if the number is not constant, then sure, this fix is better than what we had, but it's not **the** best fix (for example, in the other two similar issues I fixed yesterday, I had to drop down to Vert.x handling of the body because Vert.x already handles everything properly).

Reply:
Great, thanks